### DISCUSSION TITLE
Generating grains in 2 and 3D with a predefined function

### DISCUSSION THREAD

**Discussion by YeohWY** [neperfepx/neper#586] [Q&A] [answered]
Hello, I was able to generate 2D grains with a custom distribution function obtained from a 2D EBSD map. However, when I try to use the same function to generate 3D grains, the 3D grain diameters are much larger than the 2D counterparts.

I would like to generate 3D grain structures using a 2D manual fitted function - could you please advise on how I can achieve that? 

On a similar note, I was also looking at the different ways to generate grains in 2 and 3D - looking at a grain diameter of 0.2 mm in a 2 x 2 mm square (2D) and 2 x 2 x 2 mm cube (3D):

1. neper -T -n 100 -dim 2 -domain "square(0.002,0.002)"
2. neper -T -n 1000 -dim 3 -domain "cube(0.002,0.002,0.002)"
3. neper -T -n from_morpho -dim 2 -domain "square(0.002,0.002)" -morpho "gg(0.2e-3)"
4. neper -T -n from_morpho -dim 3 -domain "cube(0.002,0.002,0.002)" -morpho "gg(0.2e-3)"
5. neper -T -n from_morpho -dim 2 -domain "square(0.002,0.002)" -morpho "diameq:lognormal(0.2e-3,0.35)"
6. neper -T -n from_morpho -dim 3 -domain "cube(0.002,0.002,0.002)" -morpho "diameq:lognormal(0.2e-3,0.35)"

For cases 1 and 2, I divide the domain dimension by the grain area/volume to get the number of grains that should be present in the domain, and I was able to get comparable sizes between the 2 and 3D case - this was determined by looking at two point correlation functions for the grain maps (to do the TPC for the 3D grain map I just take a 2D slice of it).

In all the three methods used, the 2D grain sizes are quite consistent throughout. Whereas for the 3D case, the grain growth method generated smaller grain sizes, and the log normal method generated larger grain sizes. Is there a reason why these methods match in 2D but not in 3D?

Thank you.

**Comment by rquey** [comment, marked as answer]
To generate a 3D structure, you need 3D statistical distributions; so, you have to "convert" your 2D distributions into 3D distributions.

The second question is a bit vague, and it is difficult for me to comment without actual results/data. I can just point out that the standard deviations in the `lognormal` should be `0.00007` (=0.35*0.2e-3, as it is an absolute value), and that you should specify a sphericity distribution in cases 5 and 6; otherwise, grain shapes can become very strange.

**Reply by YeohWY** [reply]
Hi rquey, thank you for your response. In the case of 2D to 3D conversion - wouldn't the equivalent diameter of a 2D and 3D equiaxial grain be the same and therefore I should be able to directly use the 2D distributions?

**Reply by rquey** [reply]
No, they are not the same.

**Reply by mkasemer** [reply]
A 2D view of the material is a slice through the volume. As such, you will cut some grains near their centroids, giving a somewhat accurate representation of their radius as calculated from the 2D view. Other grains will be sliced near their extrema, and the radius calculated for that grain will be an underestimate of the true grain radius. As such, you should expect that your 2D distribution of grain radii will either be entirely shifted to an underestimate, or skew toward an underestimate, of what it would be if you had 3D (I seem to recall seeing a relationship for this in an old paper, though I am struggling to find it... perhaps a good short study in itself...).

Things are more difficult regarding shape... at least 3 orthogonal surface scans would be necessary to truly gain an understanding of grain shape in 3D. A single surface scan could, for instance, indicate equiaxed grains, but an orthogonal scan reveal that the grains are columnar, for example. This would of course be an extreme case, and also affect size calculations.

**Reply by northpole-pl** [reply]
Very good paper on this subject: 
_Takahashi, J., & Suito, H. (2003). Evaluation of the accuracy of the three-dimensional size distribution estimated from the schwartz-saltykov method. Metallurgical and Materials Transactions A, 34(1), 171–181._
[doi:10.1007/s11661-003-0218-6](https://doi.org/10.1007/s11661-003-0218-6)
